src/k8s-engine/wsl.ts

```diff
@@ -1137,73 +1152,105 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
         this.lastCommandComment = 'Mounting WSL data';
         await this.progressTracker.action(this.lastCommandComment, 100, this.mountData());
         this.lastCommandComment = 'Starting WSL environment';
-        await Promise.all([
+        const installerActions = [
           this.progressTracker.action(this.lastCommandComment, 100, async() => {
             const logPath = await this.wslify(paths.logs);
-            const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '').replace('/var/log', logPath);
+            const rotateConf = LOGROTATE_K3S_SCRIPT.replace(/\r/g, '')
+              .replace('/var/log', logPath);
 
             await this.writeFile('/etc/init.d/k3s', SERVICE_SCRIPT_K3S, 0o755);
+            await this.writeFile('/etc/logrotate.d/k3s', rotateConf, 0o644);
+            await this.execCommand('mkdir', '-p', '/etc/cni/net.d');
+            await this.writeFile('/etc/cni/net.d/10-flannel.conflist', FLANNEL_CONFLIST, 0o644);
+            await this.writeFile('/etc/containerd/config.toml', CONTAINERD_CONFIG, 0o644);
```

Comment:
It might be better to write the file somewhere else, and then add `--config=` to `containerd_opts` instead (so that we don't clobber the defaults, which can be useful for users as a reference).